src/devices/engo.ts

```diff
@@ -0,0 +1,310 @@
+                e.numeric("brightness", ea.STATE_SET).withUnit("%").withValueMin(10).withValueMax(100).withValueStep(10),
+                e
+                    .enum("control_algorithm", ea.STATE_SET, [
+                        "TPI_UFH",
+                        "TPI_RAD",
+                        "TPI_ELE",
+                        "HIS_02",
+                        "HIS_04",
+                        "HIS_06",
+                        "HIS_08",
+                        "HIS_10",
+                        "HIS_20",
+                        "HIS_30",
+                        "HIS_40",
+                    ])
+                    .withDescription("Control algorithm used to regulate temperature"),
+                e
+                    .numeric("frost_set", ea.STATE_SET)
+                    .withUnit("°C")
+                    .withValueMin(5)
+                    .withValueMax(17)
+                    .withValueStep(0.5)
+                    .withDescription("Frost protection setpoint"),
+                e
+                    .numeric("holiday_temp_set", ea.STATE_SET)
+                    .withUnit("°C")
+                    .withValueMin(5)
+                    .withValueMax(45)
+                    .withValueStep(0.5)
+                    .withDescription("Target temperature during holiday mode"),
+                e
+                    .numeric("holiday_days_set", ea.STATE_SET)
+                    .withUnit("day")
+                    .withValueMin(1)
+                    .withValueMax(30)
+                    .withValueStep(1)
+                    .withDescription("Number of days for holiday"),
+                e.binary("valve_protection", ea.STATE_SET, "ON", "OFF").withDescription("Prevents valve blockage during long periods of inactivity."),
+                e
+                    .enum("warm_floor", ea.STATE_SET, ["OFF", "7 min", "11 min", "15 min", "19 min", "23 min"])
+                    .withDescription("Automatically warms the floor every 60 minutes."),
+                e.enum("sensor_error", ea.STATE, ["Normal", "E1", "E2"]).withDescription("Sensor error indicator."),
+                ...tuya.exposes.scheduleAllDays(ea.STATE_SET, "HH:MM/C HH:MM/C HH:MM/C HH:MM/C HH:MM/C HH:MM/C"),
+            ];
+
+            if (options?.expose_device_state === true) {
+                exposesList.unshift(e.binary("state", ea.STATE_SET, "ON", "OFF").withDescription("Turn the thermostat ON or OFF"));
+            }
+            return exposesList;
+        },
+        meta: {
+            tuyaDatapoints: [
+                [
+                    1,
+                    "state",
+                    {
+                        to: async (value, meta) => {
+                            if (meta.options?.expose_device_state === true) {
+                                const boolValue = utils.getFromLookup(value, {on: true, off: false});
```

Comment:
```suggestion
                               return utils.getFromLookup(value, {on: true, off: false});
```

Does this also work?

Reply:
i don't think this would work as it would end the function before the await part, i have modified it in order to not use temp variable.